File: projects/code_combat/4_Backwoods_Forest/179-Arrowproof_Wolf/arrowproof.py
# You need Elemental codex 1+ to cast "Haste"
# You need special boots to Jump

hero.cast("haste", hero)
hero.moveXY(12, 34)
hero.say("Wake up, wolf!")

while True:
    item = hero.findNearestItem()
    if hero.distanceTo(item) > 12 && hero.isReady("jump"):
        hero.jumpTo(item.pos)
    else:
        hero.move(item.pos)

    if hero.canCast("haste")
        hero.cast("haste", hero)
    elif hero.isReady("reset-cooldown")
        hero.resetCooldown("haste")

File: projects/code_combat/4_Backwoods_Forest/179-Arrowproof_Wolf/arrowproof.js
// Collect mushrooms.

// First, come to the wolf pet and wake up it (say).
hero.moveXY(12, 34);
hero.say("Wake up, wolf!");
// Next collect mushrooms just usual items.
while(true) {
    var item = hero.findNearestItem();
    if (item) {
        hero.moveXY(item.pos.x, item.pos.y);
    }
}
